discovery_engine_core/web-api/openapi.yaml

```diff
@@ -0,0 +1,160 @@
+openapi: 3.0.3
+info:
+  title: Personalization API
+  version: 1.0.0-rc1
+  description: |-
+    The endpoints can be contacted at `t<name>.xaynet.dev/v1` or `t<name>.xaynet.dev/v1.0`.
+    The `v1` path will always point to the latest `v1.x`.
+    
+    # Auth
+    To authenticate with the API two bearer tokens will be provided. One will work for the
+    personalization endpoints and the other one will allow documents to be added to the system.
```

Comment:
Thanks, I'll add that

Reply:
We don't plan to generate server from this. Also the auth will be handled by the nginx in front and not by the service directly

Reply:
> The specification allows to have AND and OR for authentication.

not really 

> A declaration of which security mechanisms can be used for this operation. The list of values includes **alternative security requirement objects that can be used. Only one of the security requirement objects need to be satisfied** to authorize a request. To make security optional, an empty security requirement ({}) can be included in the array. This definition overrides any declared top-level [security](https://spec.openapis.org/oas/latest.html#oasSecurity). To remove a top-level security declaration, an empty array can be used.

Through you can could maybe define a custom security requirement object which is a combination of multiple requirements, but as far as I can tell the spec has no direct support for that (through if you use oauth2 you can require multiple scopes, but that's a different mechanism more or less only available for oath).

---

> I'm not sure I we really want to specify this here because we will be limited to authentication schema that the specification support and the service will not handle it.

If we change the auth schema it's anyway a braking change weather we define the auth as part of the schema or in the documentation.

All IANA registered http auth schemas (e.g. bearer token) are spec supported, so is a api-key in both header and query as well as extended support for mutualTLS and especially oauth2 and openIdConnect. (Through spec supported doesn't necessary mean generator supported). IMHO we shouldn't use any other auth schemas. But if we must we can just define it our self without too much problems (through it might or might not be spec conform dependent on what we speak about), or remove the security hints at this point in time.

The benefit from having the auth defined is that you can directly play around with the endpoints in the swagger editor as developer to get a feel/understanding for the API. 

Either way writing out the security schema in the spec which was already required isn't a braking change so we can always add it later.

---

> We don't plan to generate server from this. 

Through our clients might do so, and they might want to play around with the api through the swagger editor or similar OpenAPI support tooling.

Reply:
https://swagger.io/docs/specification/authentication/ under "Using Multiple Authentication Types"

Reply:
oh, nice I misread that :sweat:

Reply:
I will leave it out for the moment. We can add it later without be a break change as you said